### PR TITLE
Deprecate NCONF_WIN32() function

### PR DESCRIPTION
**Pull request by richsalz**
Deprecated undocumented (and not usable with shipped defaults) config file utility.

### CODE REVIEW

**Comment by richsalz** [comment]
needs second review?

**Comment by richsalz** [comment]
It has been a month, ping for a second review.

**Comment by mattcaswell** [comment]
This PR seemed a bit of a mystery until I read the discussion in #9263.

In the (possibly unlikely) event that someone *is* using NCONF_WIN32() what should they do if they have windows style filepaths in their config files (perhaps some guidance in the CHANGES entry)? Should their be some documentation somewhere in the man pages which tells Windows users how to put filepaths in a config file?

**Comment by richsalz** [comment]
I'm happy to add some docs, but where?  The simple answer is "convert to forward slashes" since that also works on windows.

**Comment by mattcaswell** [comment]
> I'm happy to add some docs, but where

Perhaps doc/man5/config.pod?

**Comment by richsalz** [comment]
I added a HISTORY section to doc/man5/config.pod explaining the impact, please take a look.

File: CHANGES [hunk 1]
```diff
@@ -72,6 +72,9 @@
      CMS_DEBUG_DECRYPT flag.
      [Bernd Edlinger]
 
+  *) The function NCONF_WIN32() has been deprecated.
```
Comment (mattcaswell) [inline CHANGES]: Could we expand this out more? Perhaps with a link to the man page

Comment (richsalz) [reply CHANGES]: sure, commit added.

**Comment by richsalz** [comment]
I guess @levitte needs to reconfirm.

**Comment by richsalz** [comment]
The label says it needs a second review, but this has two and is ready to merge.

**Comment by mattcaswell** [comment]
Needs reconfirmation from @levitte.

**Comment by richsalz** [comment]
@levitte are you willing to reconfirm and merge this?

**Comment by levitte** [comment]
I squashed it all into one commit before merging.  Not something I usually do, but this was generally so small that it seemed appropriate.

7cfc0a555c85220ecfd6ed038a7b859668595b72 Deprecate NCONF_WIN32() function

**Pull request closed without merging**